dashboard/agent.py

```diff
@@ -385,14 +399,16 @@ async def _check_parent():
             # Agent is failed to be started many times.
             # Push an error to all drivers, so that users can know the
             # impact of the issue.
+            gcs_publisher = None
             redis_client = ray._private.services.create_redis_client(
                 args.redis_address, password=args.redis_password)
-            gcs_publisher = None
             if args.gcs_address:
                 gcs_publisher = GcsPublisher(args.gcs_address)
+                redis_client = None
             elif gcs_pubsub_enabled():
                 gcs_publisher = GcsPublisher(
                     address=get_gcs_address_from_redis(redis_client))
+                redis_client = None
```

Comment:
How about initializing both gcs_publisher and redis_client to None, and setting redis_client only in an `else` branch here? An unnecessary initialization of redis_client can be avoided.